feature/120_pr_template

damian-romero opened the pull request:
Before merging, determine which of the templates are useful or if we want to go with the generic vs bump-version uption. The following checklist is to decide which templates we'll keep:

- [x] PULL_REQUEST_TEMPLATE/bug-pull-request.md
- [x] PULL_REQUEST_TEMPLATE/feature-pull-request.md
- [ ] PULL_REQUEST_TEMPLATE/generic-pull-request.md
- [x] PULL_REQUEST_TEMPLATE/release-pull-request.md

Commit:
Add PR templates folder and templates

PULL_REQUEST_TEMPLATE/bug-pull-request.md
PULL_REQUEST_TEMPLATE/feature-pull-request.md
PULL_REQUEST_TEMPLATE/generic-pull-request.md
PULL_REQUEST_TEMPLATE/release-pull-request.md

damian-romero commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+Bumped in:
+- [ ] README.md, [installation instructions](https://github.com/ualibraries/ReQUIAM#installation-instructions)
+- [ ] [setup.py](https://github.com/ualibraries/ReQUIAM/blob/master/setup.py)
+- [ ] [requiam/__init__.py](https://github.com/ualibraries/ReQUIAM/blob/master/requiam/__init__.py)
```
requiam/__init__.py should display as `requiam/__init__.py`
```suggestion
- [ ] [requiam/\_\_init\_\_.py](https://github.com/ualibraries/ReQUIAM/blob/master/requiam/__init__.py)
```

damian-romero commented on .github/PULL_REQUEST_TEMPLATE/feature-pull-request.md:
```diff
@@ -0,0 +1,33 @@
+
+**Test plan (If applicable)**
+<!-- Explain how you tested this feature so that others can replicate it -->
+<!--  Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->
```
```suggestion
<!-- Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->
```

damian-romero commented on .github/PULL_REQUEST_TEMPLATE/release-pull-request.md:
```diff
@@ -0,0 +1,31 @@
+Bumped in:
+- [ ] README.md, [installation instructions](https://github.com/ualibraries/ReQUIAM#installation-instructions)
+- [ ] [setup.py](https://github.com/ualibraries/ReQUIAM/blob/master/setup.py)
+- [ ] [requiam/__init__.py](https://github.com/ualibraries/ReQUIAM/blob/master/requiam/__init__.py)
```
requiam/__init__.py should display as `requiam/__init__.py`
```suggestion
- [ ] [requiam/\_\_init\_\_.py](https://github.com/ualibraries/ReQUIAM/blob/master/requiam/__init__.py)
```

damian-romero reviewed (commented):
A couple of format changes were necessary.

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+---
+name: Pull request - bug
+about: Provide specifics about a pull request for bugs
```
```suggestion
about: Provide specifics about a pull request for bug fixes
```

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+<!-- A clear and concise description of the problem the bug fix in this PR tries to solve -->
+
+<!-- Add a linked issues/pull requests -->
+See #
```
```suggestion
Fixes #
```

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+assignees: ''
+
+---
+<!-- IMPORTANT: Please do not create a Pull Request without creating an issue first. -->
```
```suggestion
<!-- IMPORTANT: Please do not create a PR without creating an issue first. -->
```

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+**Description**
+<!-- A clear and concise description of the problem the bug fix in this PR tries to solve -->
+
+<!-- Add a linked issues/pull requests -->
```
```suggestion
<!-- Add any linked issue(s) -->
```

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+<!-- How does this bug-fix address the problem? -->
+
+<!-- Add any open questions and Pre-Merge TODOs. Use checkboxes. -->
+<!-- - [ ]   -->
```
```suggestion
```

I rather have this information in the issue instead.

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+- [ ] [requiam/__init__.py](https://github.com/ualibraries/ReQUIAM/blob/master/requiam/__init__.py)
+
+*Resources*
+<!-- Links to blog posts, StackOverflow, libraries or add-ons used to solve this problem -->
```
```suggestion
```

I prefer that this stays in the issue information.

astrochun commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,41 @@
+---
+name: Pull request - bug
```
```suggestion
name: Bug
```

This is the name that shows up under PR page. It's redundant as a result

astrochun commented on .github/PULL_REQUEST_TEMPLATE/release-pull-request.md:
```diff
@@ -0,0 +1,31 @@
+- A #issue
+- B #issue
+- C #issue
+- n  
```
Not sure what you're trying to do here. For releases, we want to auto close issues.

damian-romero replied:
```suggestion
<!-- A clear and concise description of the updates contained in this release. Example:  -->
<!-- Incorporates Travis CI feature #91, summary for script_run #99, and root portal handling #88 -->

- Fixes #
- Fixes #
- Fixes #
- Fixes #
```

astrochun reviewed (commented):
Accepted @damian-romero suggestions

Additional changes incorporated.

In the future, start with a simple generic one so we can improve. A typo in one results in four typos to fix.

damian-romero committed:
Apply suggestions from code review #1

Fix typos and content

Co-authored-by: Chun Ly <astro.chun@gmail.com>

damian-romero commented on .github/PULL_REQUEST_TEMPLATE/feature-pull-request.md:
```diff
@@ -0,0 +1,33 @@
+---
+name: Pull request - feature
```
```suggestion
Feature
```

damian-romero commented on .github/PULL_REQUEST_TEMPLATE/bug-pull-request.md:
```diff
@@ -0,0 +1,33 @@
+<!--Fields in **bold** are REQUIRED, fields in *italics* are OPTIONAL -->
+
+**Description**
+<!-- A clear and concise description of the problem the bug fix in this PR tries to solve -->
```
```suggestion
<!-- A clear and concise description of the problem the bug that this PR tries to solve -->
```

Commit:
Fix further typos and content in PR templates

astrochun committed:
Improvements to PR templates

 - Simplify title and about description
 - Use relative paths for files (refer to specific branch instead of master)
 - Change Description's focus to get to the point (issue tickets are used to track the problem)
 - Testing optional since some bug fixes might require testing (e.g., docs)

astrochun committed:
Delete generic-pull-request.md template

damian-romero reviewed (commented):
I think they're set
